tests/testthat/test-odbc-config.R

```diff
@@ -0,0 +1,64 @@
+test_that("odbcListConfig returns appropriate result", {
+  skip_on_os(c("windows", "solaris"))
+  skip_if(!has_odbc(), "odbcinst not available.")
+
+  res <- odbcListConfig()
+
+  expect_type(res, "character")
+  expect_length(res, 3)
+  expect_named(res, c("drivers", "system_dsn", "user_dsn"))
+  expect_match(res, "\\.ini")
+})
+
+test_that("odbcListConfig returns an empty vector on Windows", {
+  skip_on_os(c("mac", "linux", "solaris"))
+
+  res <- odbcListConfig()
+
+  expect_equal(res, character(0))
+})
+
+test_that("odbcListConfig errors informatively without unixODBC", {
+  skip_on_os(c("windows", "solaris"))
```

Comment:
FWIW I'd be tempted to make a `is_window()` function so that you could mock it and then drop the skips.

Reply:
Sure thing. In these tests I always skip on Solaris. Should `is_windows()` on Solaris return true? Or figure out how to support Solaris and test on those systems as well?

Reply:
Once you've mocked the right functions, it shouldn't matter which OS it's actually running on.

And I'm reasonably certain we don't generally need to worry about solaris any more?

Reply:
[99938dd](https://github.com/r-dbi/odbc/pull/680/commits/99938dd10532cd679075dece04ff61bdfb29ecfb) and [e40215e](https://github.com/r-dbi/odbc/pull/680/commits/e40215ef3a190088577d121b8a584891f4bcd938) should get us most of the way there, though I'm not sure how we'd mock through that first test where we do need odbcinst output.